src/AppInstallerCLICore/Command.cpp

```diff
@@ -43,6 +44,21 @@ namespace AppInstaller::CLI
         {
             m_fullName = name;
         }
+
+        if (!aliases.empty())
+        {
+            for (auto& alias : aliases)
+            {
+                std::string _aliasFullName;
+                if (!parent.empty())
+                {
+                    _aliasFullName = parent;

```

Comment:
What if the parent is also aliased?

Reply:
Are you suggesting that the entire tree needs to be expanded to find all potential alias full names?

Reply:
I'm suggesting that we need to give some thought to that case. Depending on how we use the full name, we could need to generate all possible full names, or just one, or maybe we don't need the alias full names at all.

I *think* we only use the full name for logging? If that's the case we probably don't need to compute the alias full names, but we'd need to verify.

Reply:
I did have to use the full name to ensure that there weren't collisions between aliases and commands, given that aliases can be the same but not collide if they have different parents. Since the collision check is recursive, I don't think the full expansion is needed. By checking that `root/search` and `root/find` don't collide with any other commands, then any sub-commands underneath them will also not collide with commands in other branches. So, as long as there are no aliases that collide at the same level, checking only one branch of the sub-trees suffices as a check for all other branches within that path.